src/Sarif.UnitTests/SarifCurrentToVersionOneTests.cs

```diff
@@ -198,5 +185,208 @@ public void SarifTransformerTests_ToVersionOne_OneRunWithLogicalLocations()
+  ""version"": ""2.0.0"",
+  ""runs"": [
+    {
+      ""tool"": {
+        ""name"": ""CodeScanner"",
+        ""semanticVersion"": ""2.1.0""
+      },
+      ""invocations"": [
+        null
+      ],
+      ""results"": [],
+      ""resources"": {
+        ""rules"": {
+          ""C2001"": {
+            ""id"": ""C2001"",
+            ""shortDescription"": {
+              ""text"": ""A variable was used without being initialized.""
+            },
+            ""messageStrings"": {
+              ""default"": ""Variable \""{0}\"" was used without being initialized.""
+            },
+            ""properties"": {
+              ""some_key"": ""FoxForceFive""
+            }
+          },
+          ""C2002"": {
+            ""id"": ""C2002"",
+            ""fullDescription"": {
+              ""text"": ""Catfish season continuous hen lamb include dose copy grant.""
+            },
+            ""configuration"": {
+              ""enabled"": true,
+              ""defaultLevel"": 2
+            },
+            ""helpLocation"": {
+              ""uri"": ""http://www.domain.com/rules/c2002.html""
+            }
+          },
+          ""C2003"": {
+            ""id"": ""C2003"",
+            ""name"": {
+              ""text"": ""Rule C2003""
+            },
+            ""shortDescription"": {
+              ""text"": ""Rules were meant to be broken.""
+            },
+            ""fullDescription"": {
+              ""text"": ""Rent internal rebellion competence biography photograph.""
+            },
+            ""configuration"": {}
+          }
+        }
+      }
+    }
+  ]
+}";
+            const string V1LogExpectedText =
+@"{
+  ""$schema"": ""http://json.schemastore.org/sarif-1.0.0"",
+  ""version"": ""1.0.0"",
+  ""runs"": [
+    {
+      ""tool"": {
+        ""name"": ""CodeScanner"",
+        ""semanticVersion"": ""2.1.0""
+      },
+      ""results"": [],
+      ""rules"": {
+        ""C2001"": {
+          ""id"": ""C2001"",
+          ""shortDescription"": ""A variable was used without being initialized."",
+          ""messageFormats"": {
+            ""default"": ""Variable \""{0}\"" was used without being initialized.""
+          },
+          ""properties"": {
+            ""some_key"": ""FoxForceFive""
+          }
+        },
+        ""C2002"": {
+          ""id"": ""C2002"",
+          ""fullDescription"": ""Catfish season continuous hen lamb include dose copy grant."",
+          ""configuration"": ""enabled"",
+          ""defaultLevel"": ""error"",
+          ""helpUri"": ""http://www.domain.com/rules/c2002.html""
+        },
+        ""C2003"": {
+          ""id"": ""C2003"",
+          ""name"": ""Rule C2003"",
+          ""shortDescription"": ""Rules were meant to be broken."",
+          ""fullDescription"": ""Rent internal rebellion competence biography photograph."",
+          ""configuration"": ""disabled"",
+          ""defaultLevel"": ""pass""
+        }
+      }
```

Comment:
As in the case of the `file` object, add a test that shows that _new_ v2 `rule` properties go in the v1 `rule`'s property bag. For example, `helpLocation` and `configuration.parameters`. Note that the latter is itself a property bag.